### PR TITLE
Add scripts/upload_source.py

### PR DESCRIPTION
**Pull request by dcolligan**
This is a script to easily upload a version of the server to pypi

Also do some refactoring

### CODE REVIEW

File: scripts/auth.yml.example [hunk 1]
```diff
@@ -1,2 +1,5 @@
 google:
     key: YOUR_KEY
+pypi:
```
Comment (jeromekelleher) [inline scripts/auth.yml.example]: This makes me nervous (accidental commit of credentials to github) - does ~/.pypirc not work with twine?

**Comment by jeromekelleher** [comment]
This seems like overkill to me also I'm afraid. It'll take a few seconds to run
```
python setup.py sdist
twine upload  dist/ga4gh-{version}.tar.gz
```
This will happen (say) once a month, and doesn't need to be automated. (Plus, I might not actually want to do the things that are in the script --- what if I want to keep a bunch of earlier versions in my dist directory for testing?)

The refactorings look good --- feel free to keep those. I'm -1 on the script itself though.

File: scripts/auth.yml.example (hunk 1 above)
Comment (dcolligan) [inline scripts/auth.yml.example]: It looks like it does work with pypirc, fyi:

https://github.com/pypa/twine/blob/master/twine/utils.py

**Comment by dcolligan** [comment]
@jeromekelleher  Alright, took out the upload script but kept the rest of the changes

**Comment by dcolligan** [comment]
(although I would note a script like this might be useful to enforce some kind of pushing policy if/when we tackle #228 )

**Comment by jeromekelleher** [comment]
Thanks @dcolligan, +1.

**Pull request merged**